script.js

```diff
@@ -0,0 +1,112 @@
+    }]
+    }
+    
+
+
+//Loop for repeat of card creation
+for (let xy = 0; xy < 6; xy++){
+    //creation step
+    let cloned_clone = document.getElementsByClassName("card_clone")[0].cloneNode(true)
+    document.getElementById("frame43").appendChild(cloned_clone)
+
+        
+    //card filling info with card_info array
+        document.getElementsByClassName("desktop_card_img_class")[xy].src = card_info.image[xy]
+        document.getElementsByClassName("desktop_card_title")[xy].innerHTML = card_info.name[xy]
+        document.getElementsByClassName("card_secondary")[xy].innerHTML = card_info.description[xy]
+        document.getElementsByClassName("card_btns1")[xy].innerHTML = card_info.technologies[0].tech_stack[0]
+        document.getElementsByClassName("card_btns2")[xy].innerHTML = card_info.technologies[0].tech_stack[1]
+        document.getElementsByClassName("card_btns3")[xy].innerHTML = card_info.technologies[0].tech_stack[2]
+
+    
+        //add event listener to each card's button for the popup and fill popup info
+        document.getElementsByClassName('action_btns')[xy].addEventListener('click', function popup_manager () {
+        
+            document.getElementById("popup-container").style.display = "flex"
+        
+            document.getElementById("p-button-container-text").innerHTML = card_info.description[xy]
+            document.getElementById("img-project").src = card_info.image[xy]
+            document.getElementById("popup-title").innerHTML = card_info.name[xy]
+            document.getElementById("live-link").href = card_info.live_link[xy]
+            document.getElementById("source-link").href = card_info.source_link[xy]
+        
+            document.getElementById('popup-tag1').innerHTML = card_info.technologies[0].tech_stack[0]
+            document.getElementById('popup-tag2').innerHTML = card_info.technologies[0].tech_stack[1]
+            document.getElementById('popup-tag3').innerHTML = card_info.technologies[0].tech_stack[2]
+        
+        })
+
+}
+
+
+
+
+//x icon on popup container to close
+document.getElementById("cancel-icon").addEventListener('click', function() {
+    document.getElementById("popup-container").style.display = "none"
+
+})
+
+
+let form_error
+
+document.getElementsByClassName("form_email")[0].addEventListener('blur', function() {
+    let lowercased = document.getElementsByClassName("form_email")[0].value.toLowerCase()
+    if (document.getElementsByClassName("form_email")[0].value != lowercased) {
+        document.getElementById('form_email_error').innerHTML = "Use lower caps for email input"
+        form_error = true
+    }
+    else {
+        document.getElementById('form_email_error').innerHTML = ""
+        form_error = false
+    }
+})
+
+document.getElementById('form').addEventListener('submit', function(q) {
+    q.preventDefault()
+    if (form_error != true) {
+        document.getElementById("form_submit").submit();
+    }
```

Comment:
- [ ] Great effort working on the `form validation` 👋 👋. I would kindly suggest we stick to the [project requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/9_validate_form.md) as much as possible.
I could notice the functionality you added to your form is functioning in such a way that the `error message` popped up the moment a user wrote a wrong `email address` i.e `has uppercase` even before submitting the form.
I would kindly suggest you add the functionality in such a way that the `error message` will only pop up when the user tries to submit the form with a wrong `email address`

> N.B: The contact form contains at least one validation checking if the email is written in all lowercase:
.If a user submits the contact form with the email field containing uppercase characters an error message should show up.
.If a user submits the contact form with the email field containing only lowercase characters the form is sent normally.

Reply:
This sounds like intentionally obtuse user experience but ill change it from "blur" to "submit"

Reply:
Done - it now only shows when the button is clicked and the criteria is met